src/braket/pennylane_plugin/braket_device.py

```diff
@@ -358,15 +371,32 @@ def batch_execute(self, circuits, **run_kwargs):
             **self._run_kwargs,
         )
         # Call results() to retrieve the Braket results in parallel.
-        braket_results_batch = task_batch.results(
-            fail_unsuccessful=True, max_retries=self._max_retries
-        )
-
-        if self.tracker.active:
-            batch_len = len(circuits)
-            total_shots = batch_len * batch_shots
-            self.tracker.update(batches=1, executions=batch_len, shots=total_shots)
-            self.tracker.record()
+        try:
+            braket_results_batch = task_batch.results(
+                fail_unsuccessful=True, max_retries=self._max_retries
+            )
+
+        # Update the tracker before raising an exception further if some circuits do not complete.
+        finally:
+            if self.tracker.active:
+                for task in task_batch.tasks:
+                    tracking_data = {}
+                    tracking_data["braket_task_id"] = task.id
+                    try:
+                        simulation_ms = (
+                            task.result().additional_metadata.simulatorMetadata.executionDuration
+                        )
+                        tracking_data["braket_simulator_ms"] = simulation_ms
+                        tracking_data["braket_simulator_billed_ms"] = max(
+                            simulation_ms, MIN_SIMULATOR_BILLED_MS
+                        )
```

Comment:
This a duplicate of the code above; might be worth extracting into a common private method.